Lib/test/test_urllib2.py

```diff
@@ -818,19 +810,20 @@ def test_file(self):
         o = h.parent = MockOpener()
 
         TESTFN = os_helper.TESTFN
-        urlpath = sanepathname2url(os.path.abspath(TESTFN))
         towrite = b"hello, world\n"
+        canonurl = 'file:' + urllib.request.pathname2url(os.path.abspath(TESTFN))
+        parsed = urlparse(canonurl)
         urls = [
-            "file://localhost%s" % urlpath,
-            "file://%s" % urlpath,
-            "file://%s%s" % (socket.gethostbyname('localhost'), urlpath),
+            canonurl,
+            parsed._replace(netloc='localhost').geturl(),
```

Comment:
How would this work on Windows if the current directory is a share ("\\server\share\...")?

Reply:
Thanks - the test is now skipped in that case.